src/Periphery/LiFiDEXAggregator.sol

```diff
@@ -732,6 +748,151 @@ contract LiFiDEXAggregator is WithdrawablePeriphery {
+            if (vault == address(0)) revert InvalidCallData();
+            if (from == msg.sender) {
+                IERC20(tokenIn).safeTransferFrom(msg.sender, vault, amountIn);
+            } else if (from == address(this)) {
+                IERC20(tokenIn).safeTransfer(vault, amountIn);
+            }
+            ISyncSwapVault(vault).deposit(tokenIn, pool);
+        } else {
+            if (from == msg.sender) {
+                IERC20(tokenIn).safeTransferFrom(msg.sender, pool, amountIn);
+            } else if (from == address(this)) {
+                IERC20(tokenIn).safeTransfer(pool, amountIn);
+            }
+        }
+        bytes memory data = abi.encode(tokenIn, to, withdrawMode);
+        ISyncSwapPool(pool).swap(data, from, address(0), new bytes(0));
+    }
+
+    /// @dev Common logic for iZiSwap callbacks
+    /// @param amountToPay The amount of tokens to be sent to the pool
+    /// @param data The data passed through by the caller
+    function _handleIzumiV3SwapCallback(
+        uint256 amountToPay,
+        bytes calldata data
+    ) private {
+        if (msg.sender != lastCalledPool) {
+            revert IzumiV3SwapCallbackUnknownSource();
+        }
+
+        address tokenIn = abi.decode(data, (address));
+
+        if (amountToPay <= 0) {
+            revert IzumiV3SwapCallbackNotPositiveAmount();
+        }
+
+        lastCalledPool = IMPOSSIBLE_POOL_ADDRESS;
+        IERC20(tokenIn).safeTransfer(msg.sender, amountToPay);
+    }
+
+    /// @notice Called to `msg.sender` after executing a swap via IiZiSwapPool#swapX2Y
+    /// @dev In the implementation you must pay the pool tokens owed for the swap
+    /// @dev The caller of this method must be checked to be an iZiSwap pool deployed by the canonical iZiSwap factory
+    /// @param amountX The amount of tokenX that must be sent to the pool by the end of the swap
+    /// @param amountY The amount of tokenY that was sent by the pool in the swap
+    /// @param data Any data passed through by the caller via the IiZiSwapPool#swapX2Y call
+    function swapX2YCallback(
+        uint256 amountX,
+        // solhint-disable-next-line no-unused-vars
+        uint256 amountY,
+        bytes calldata data
+    ) external {
+        _handleIzumiV3SwapCallback(amountX, data);
+    }
+
+    /// @notice Called to `msg.sender` after executing a swap via IiZiSwapPool#swapY2X
+    /// @dev In the implementation you must pay the pool tokens owed for the swap
+    /// @dev The caller of this method must be checked to be an iZiSwap pool deployed by the canonical iZiSwap factory
+    /// @param amountX The amount of tokenX that was sent by the pool in the swap
+    /// @param amountY The amount of tokenY that must be sent to the pool by the end of the swap
+    /// @param data Any data passed through by the caller via the IiZiSwapPool#swapY2X call
+    function swapY2XCallback(
+        // solhint-disable-next-line no-unused-vars
+        uint256 amountX,
```

Comment:
we can remove the name of the variable and the solhint comment

Reply:
+1